server.go

```diff
@@ -836,11 +836,23 @@ func (sp *serverPeer) OnGetCFHeaders(_ *peer.Peer, msg *wire.MsgGetCFHeaders) {
 		headersMsg.AddCFHeader(&header)
 	}
 
-	headersMsg.Extended = msg.Extended
+	headersMsg.FilterType = msg.FilterType
 	headersMsg.StopHash = hashList[len(hashList)-1]
 	sp.QueueMessage(headersMsg, nil)
 }
 
+// OnGetCFTypes is invoked when a peer receives a getcftypes bitcoin message.
+func (sp *serverPeer) OnGetCFTypes(_ *peer.Peer, msg *wire.MsgGetCFTypes) {
+	// Ignore getcftypes requests if cfg.NoCFilters is set or we're not in
+	// sync.
+	if cfg.NoCFilters || !sp.server.blockManager.IsCurrent() {
+		return
+	}
+
+	cfTypesMsg := wire.NewMsgCFTypes([]uint8{0, 1})
```

Comment:
A TODO should be added here to query some information within the peer as which types the peer _actually_ offers.

Reply:
I've added this TODO. Right now, our code only offers both types or neither, but this could be configurable in the future.